monai/data/readers/niftireader.py

```diff
@@ -97,4 +103,14 @@ def __getitem__(self, index):
             random.seed(seed)  # ensure randomized transforms roll the same values for segmentations as images
             seg = self.seg_transform(seg)
 
-        return img, seg
+        if self.image_only or meta_data is None:
+            return img, seg
+
+        compatible_meta = {}
```

Comment:
to get the corresponding input file name when saving a segmentation output, I need to change the current `NiftiDataset ` class to return metadata. What's implemented here is a quick workaround. 

this is related to the input pipeline design, any comments/suggestions? @ericspod @atbenmurray @Nic-Ma  

dtypes of some of the metadata items are not supported by the default collate function of pytorch dataloader, I have to put a filter here to return only the compatbile metadata.

Reply:
@wyli #74  also uses metadata to save original affine matrix. This is used to reverse the canonical transformation. I'll keep this in mind in case pipeline changes.